File: java/hamming/src/main/java/Hamming.java
public class Hamming {

  String string1;
  String string2;

  public Hamming(String string1, String string2){
    if (string1.length() != string2.length()){
      throw new IllegalArgumentException();
    }
    this.string1 = string1;
    this.string2 = string2;

  }

  public int getHammingDistance(){

    if (string1.equals(string2)){
      return 0;
    }
    int count = 0;

    for (int i = 0; i < string1.length(); i++){
      if (string1.charAt(i) != string2.charAt(i)){
        count++;
      }
    }
    return count;
  }

}

File: ruby/hamming/hamming.rb
class Hamming

  def self.compute(string_1, string_2)

    if string_1.length != string_2.length
      raise(ArgumentError)
      return
    end

    differences_counter = 0
    index_counter = 0

    while index_counter < string_1.length
      if string_1[index_counter] != string_2[index_counter]
      differences_counter +=1
      end
      index_counter +=1
    end


    return differences_counter

  end

end